src/bootstrap/defaults/bootstrap.dist.toml

```diff
@@ -13,9 +13,6 @@ extended = true
 download-ci-llvm = false
 
 [rust]
-# We have several defaults in bootstrap that depend on whether the channel is `dev` (e.g. `omit-git-hash` and `download-ci-llvm`).
-# Make sure they don't get set when installing from source.
-channel = "nightly"
```

Comment:
This was added in https://github.com/rust-lang/rust/pull/125181, but since source tarballs use the "dist" profile by default, it conflicts with the behavior of treating `src/ci/channel` as the default channel value when the user doesn't set it manually.

Reply:
I assume most (if not all?) distro workflows prefer tarball sources and bootstrap uses `src/ci/channel` as the default for tarballs. So the "dev" channel will be used when users choose the git source and no channel is explicitly set.

Reply:
How did this even work on beta/stable? They override `src/ci/channel`, but they also use the `dist` profile, so by the reasoning that you wrote here, shouldn't that mean that they were being overridden as `nightly`?

(Btw, we should probably separate the CI and tarball/dist profiles to avoid these kinds of conflicts.)

Reply:
They work on beta/stable because they are running on Git source not tarball.

Reply:
Oh wait, there should be some additional logic as well, otherwis, it will get overridden just like the tarball source.

Reply:
With https://github.com/rust-lang/rust/pull/139574/commits/9847c738930b329f70e34d97048a35cd795654b8 this isn't relevant anymore.